[Bug]:  LocalNameStandardizer not handling Integer1Type, Integer127Type

### What happened?

when applying localSplitter, TypeAssigner, then LocalNameStandardiser. It fails as it doesn't handle Integer1Type, Integer127Type types etc.

### Version

Latest develop branch

### Relevant log output

```shell
Exception in thread "main" java.lang.IllegalStateException: Failed to apply sootup.core.transform.RunTimeBodyInterceptor@7ec4648f to <org.mockito.internal.reporting.SmartPrinter: boolean isMultiLine(org.mockito.invocation.MatchableInvocation,java.util.List)>
	at sootup.java.bytecode.frontend.conversion.AsmMethodSource.resolveBody(AsmMethodSource.java:241)
	at sootup.core.model.SootMethod.lazyBodyInitializer(SootMethod.java:98)
	at com.google.common.base.Suppliers$NonSerializableMemoizingSupplier.get(Suppliers.java:200)
	at sootup.core.model.SootMethod.getBody(SootMethod.java:177)
	at upb.thesis.SetUp.lambda$1(SetUp.java:190)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at upb.thesis.SetUp.lambda$0(SetUp.java:188)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.HashMap$ValueSpliterator.forEachRemaining(HashMap.java:1779)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at upb.thesis.SetUp.setupSootUp(SetUp.java:187)
	at upb.thesis.SetUp.executeStaticAnalysis(SetUp.java:48)
	at upb.thesis.Main.main(Main.java:80)
Caused by: java.lang.UnsupportedOperationException
	at sootup.interceptors.typeresolving.types.AugmentIntegerTypes$Integer1Type.accept(AugmentIntegerTypes.java:63)
	at sootup.interceptors.typeresolving.types.AugmentIntegerTypes$Integer1Type.accept(AugmentIntegerTypes.java:49)
	at sootup.core.jimple.basic.LocalGenerator.generateLocal(LocalGenerator.java:68)
	at sootup.interceptors.LocalNameStandardizer.interceptBody(LocalNameStandardizer.java:77)
	at sootup.core.transform.RunTimeBodyInterceptor.interceptBody(RunTimeBodyInterceptor.java:54)
	at sootup.java.bytecode.frontend.conversion.AsmMethodSource.resolveBody(AsmMethodSource.java:235)
	... 18 more
```

--- a/sootup.interceptors/src/main/java/sootup/interceptors/LocalNameStandardizer.java
+++ b/sootup.interceptors/src/main/java/sootup/interceptors/LocalNameStandardizer.java
@@ -32,7 +32,6 @@ import sootup.core.transform.BodyInterceptor;
 import sootup.core.types.Type;
 import sootup.core.views.View;
 
-// https://github.com/Sable/soot/blob/master/src/main/java/soot/jimple/toolkits/scalar/LocalNameStandardizer.java
 
 /** @author Zun Wang */
 public class LocalNameStandardizer implements BodyInterceptor {

--- a/sootup.interceptors/src/main/java/sootup/interceptors/LocalNameStandardizer.java
+++ b/sootup.interceptors/src/main/java/sootup/interceptors/LocalNameStandardizer.java
@@ -32,7 +32,6 @@ import sootup.core.transform.BodyInterceptor;
 import sootup.core.types.Type;
 import sootup.core.views.View;
 
-
 /** @author Zun Wang */
 public class LocalNameStandardizer implements BodyInterceptor {
 

--- a/sootup.interceptors/src/main/java/sootup/interceptors/LocalNameStandardizer.java
+++ b/sootup.interceptors/src/main/java/sootup/interceptors/LocalNameStandardizer.java
@@ -29,12 +29,8 @@ import sootup.core.jimple.basic.LocalGenerator;
 import sootup.core.jimple.common.stmt.Stmt;
 import sootup.core.model.Body;
 import sootup.core.transform.BodyInterceptor;
-import sootup.core.types.PrimitiveType;
 import sootup.core.types.Type;
 import sootup.core.views.View;
-import sootup.interceptors.typeresolving.types.AugmentIntegerTypes;
-import sootup.interceptors.typeresolving.types.BottomType;
-import sootup.interceptors.typeresolving.types.TopType;
 
 // https://github.com/Sable/soot/blob/master/src/main/java/soot/jimple/toolkits/scalar/LocalNameStandardizer.java
 
@@ -70,20 +66,6 @@ public class LocalNameStandardizer implements BodyInterceptor {
       Local local = iterator.next();
       Local newLocal;
       Type type = local.getType();
-
-      if (type instanceof BottomType || type instanceof TopType) {
-        // TODO: log that likely the jimple is not formed correctly
-        // TODO: handle module signatures
-        type = view.getIdentifierFactory().getClassType("java.lang.Object");
-      }
-      if (type instanceof AugmentIntegerTypes.Integer1Type) {
-        type = PrimitiveType.BooleanType.getInstance();
-      }
-      if (type instanceof AugmentIntegerTypes.Integer127Type
-          || type instanceof AugmentIntegerTypes.Integer32767Type) {
-        type = PrimitiveType.IntType.getInstance();
-      }
-
       newLocal = lgen.generateLocal(type);
       builder.replaceLocal(local, newLocal);
     }

--- a/sootup.interceptors/src/main/java/sootup/interceptors/LocalNameStandardizer.java
+++ b/sootup.interceptors/src/main/java/sootup/interceptors/LocalNameStandardizer.java
@@ -29,9 +29,12 @@ import sootup.core.jimple.basic.LocalGenerator;
 import sootup.core.jimple.common.stmt.Stmt;
 import sootup.core.model.Body;
 import sootup.core.transform.BodyInterceptor;
+import sootup.core.types.PrimitiveType;
 import sootup.core.types.Type;
 import sootup.core.views.View;
+import sootup.interceptors.typeresolving.types.AugmentIntegerTypes;
 import sootup.interceptors.typeresolving.types.BottomType;
+import sootup.interceptors.typeresolving.types.TopType;
 
 // https://github.com/Sable/soot/blob/master/src/main/java/soot/jimple/toolkits/scalar/LocalNameStandardizer.java
 
@@ -68,10 +71,17 @@ public class LocalNameStandardizer implements BodyInterceptor {
       Local newLocal;
       Type type = local.getType();
 
-      if (type instanceof BottomType) {
+      if (type instanceof BottomType || type instanceof TopType) {
         // TODO: log that likely the jimple is not formed correctly
         // TODO: handle module signatures
         type = view.getIdentifierFactory().getClassType("java.lang.Object");
+      }
+      if (type instanceof AugmentIntegerTypes.Integer1Type) {
+        type = PrimitiveType.BooleanType.getInstance();
+      }
+      if (type instanceof AugmentIntegerTypes.Integer127Type
+          || type instanceof AugmentIntegerTypes.Integer32767Type) {
+        type = PrimitiveType.IntType.getInstance();
       }
 
       newLocal = lgen.generateLocal(type);

--- a/sootup.interceptors/src/main/java/sootup/interceptors/typeresolving/TypeResolver.java
+++ b/sootup.interceptors/src/main/java/sootup/interceptors/typeresolving/TypeResolver.java
@@ -81,7 +81,8 @@ public class TypeResolver {
         new TypePromotionVisitor(builder, evalFunction, hierarchy);
     typings = typings.stream().map(promotionVisitor::getPromotedTyping).collect(Collectors.toSet());
 
-    // Promote `null`/`BottomType` types to `Object`.
+    // Promote `null`/`BottomType`/'TopType' types to `Object`, and other types which have
+    // UnsupportedOperation in TypePromotionVisitor.
     for (Typing typing : typings) {
       for (Local local : locals) {
         typing.set(local, convertUnderspecifiedType(typing.getType(local)));
@@ -353,6 +354,12 @@ public class TypeResolver {
       // It is probably possible to use the debug information to choose a type here, but that
       // complexity is not worth it for such an edge case.
       return objectType;
+    } else if (type instanceof AugmentIntegerTypes.Integer1Type) {
+      return PrimitiveType.getBoolean();
+    } else if (type instanceof AugmentIntegerTypes.Integer127Type) {
+      return PrimitiveType.getByte();
+    } else if (type instanceof AugmentIntegerTypes.Integer32767Type) {
+      return PrimitiveType.getShort();
     } else {
       return type;
     }

